src/std/misc/string.ss

```diff
@@ -215,3 +219,32 @@
 	(string-set! str i (random-word-char)))
       str)
     ""))
+
+;; str converts all of its arguments into a single string.
+;; When called without an argument an empty string is returned.
+;;
+;; Examples:
+;;  (str 2.0)               => "2.0"
+;;  (str "hello" ", world") => "hello, world"
+(def* str
+  ((v) (str1 v))
+  (xs  (call-with-output-string
+         (lambda (port)
+           (let loop ((rest xs))
+             (when (pair? rest)
+               (str1 (car rest) port)
+               (loop (cdr rest))))))))
+
+(def (str1 v (port #f))
+  (def (representable? obj)
+    (def (repr-obj? v) (and (object? v) (find-method (object-type v) ':pr)))
+    (any-of [list? hash-table? vector? ##values? repr-obj?] obj))
```

Comment:
I would reify this in a list outside the body of `str`, otherwise we'll be allocating a list every time.

Reply:
And it's even better to write it as `(? (or list? hash-table? vector? repr-obj?)) obj)`